module/zfs/spa.c

```diff
@@ -6497,7 +6497,8 @@ spa_tryimport(nvlist_t *tryconfig)
 	 * Create and initialize the spa structure.
 	 */
 	char *name = kmem_alloc(MAXPATHLEN, KM_SLEEP);
-	(void) snprintf(name, MAXPATHLEN, "%s%s", TRYIMPORT_NAME, poolname);
+	(void) snprintf(name, MAXPATHLEN, "%s-%s-%llx",
+	    TRYIMPORT_NAME, poolname, (u_longlong_t)curthread);
```

Comment:
To ensure that we have uniqueness, we probably want the format to be `TRYIMPORT_NAME-curthread-poolname`. This way if the poolname is very long (i.e. close in length to MAXPATHLEN), that we can still have a unique name.

Reply:
good point! fixed